rct229/rulesets/ashrae9012019/section5/section5rule1.py

```diff
@@ -0,0 +1,280 @@
+    """Rule 1 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule1, self).__init__(
+            rmrs_used=produce_ruleset_model_instance(
+                USER=True,
+                BASELINE_0=True,
+                BASELINE_90=True,
+                BASELINE_180=True,
+                BASELINE_270=True,
+                PROPOSED=True,
+            ),
+            each_rule=Section5Rule1.RMDRule(),
+            index_rmr=BASELINE_0,
+            id="5-1",
+            description="There are four baseline rotations (i.e., four baseline models differing in azimuth by 90 degrees and four sets of baseline model results) if vertical fenestration area per each orientation differ by more than 5%.",
+            ruleset_section_title="Envelope",
+            standard_section="Table G3.1#5a baseline column",
+            is_primary_rule=True,
+            list_path="ruleset_model_descriptions[0]",
+        )
+
+    class RMDRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule1.RMDRule, self).__init__(
+                rmrs_used=produce_ruleset_model_instance(
+                    USER=True,
+                    BASELINE_0=True,
+                    BASELINE_90=True,
+                    BASELINE_180=True,
+                    BASELINE_270=True,
+                    PROPOSED=True,
+                ),
+                each_rule=Section5Rule1.RMDRule.BuildingRule(),
+                index_rmr=BASELINE_0,
+                list_path="$.buildings[*]",
+            )
+
+        def create_data(self, context, data):
+            rmd_b0 = context.BASELINE_0
+            rmd_b90 = context.BASELINE_90
+            rmd_b180 = context.BASELINE_180
+            rmd_b270 = context.BASELINE_270
+            rmd_p = context.PROPOSED
+            rmd_u = context.USER
+
+            has_proposed = True if find_one("$.type", rmd_p) else False
+            has_user = True if find_one("$.type", rmd_u, False) else False
+
+            baseline_rmd_list = [rmd_b0, rmd_b90, rmd_b180, rmd_b270]
+            rmd_list = [rmd_b0, rmd_b90, rmd_b180, rmd_b270, rmd_p, rmd_u]
+
+            # count the rmds that aren't None
+            no_of_rmds = len(
+                filter_(
+                    rmd_list,
+                    lambda rmd: rmd is not None,
+                )
+            )
+
+            # filter out baseline rmds that have the type key
+            baseline_list = [
+                find_one("$.type", rmd_b)
+                for rmd_b in baseline_rmd_list
+                if find_one("$.type", rmd_b, False)
+            ]
+
+            no_of_output_instance = len(baseline_list)
+
+            return {
+                "has_proposed": has_proposed,
+                "has_user": has_user,
+                "baseline_list": baseline_list,
+                "no_of_rmds": no_of_rmds,
+                "no_of_output_instance": no_of_output_instance,
+            }
```

Comment:
After review this rule, I feel that you do not need this `create_data` function. All info can be retrieved inside `BuildingRule`